Code/Framework/AzToolsFramework/Tests/FocusMode/EditorFocusModeTests.cpp

```diff
@@ -0,0 +1,128 @@
+            m_entityMap["passengerId1"] =   CreateEditorEntity("Passenger", m_entityMap["carId"]);
+            m_entityMap["sportsCarId"] =    CreateEditorEntity("SportsCar", m_entityMap["streetId"]);
+            m_entityMap["passengerId2"] =   CreateEditorEntity("Passenger", m_entityMap["sportsCarId"]);
+        }
+
+        AZ::EntityId CreateEditorEntity(const char* name, AZ::EntityId parentId)
+        {
+            AZ::Entity* entity = nullptr;
+            UnitTest::CreateDefaultEditorEntity(name, &entity);
+
+            // Parent
+            AZ::TransformBus::Event(entity->GetId(), &AZ::TransformInterface::SetParent, parentId);
+
+            return entity->GetId();
+        }
+
+        void TearDown() override
+        {
+            m_app.Stop();
+        }
+
+        UnitTest::ToolsTestApplication m_app{ "EditorFocusModeTests" };
+        AZ::ComponentApplication::Descriptor m_descriptor;
+        AZStd::unordered_map<AZStd::string, AZ::EntityId> m_entityMap;
+    };
+
+    TEST_F(EditorFocusModeTests, EditorFocusModeTests_SetFocus)
+    {
+        FocusModeInterface* focusModeInterface = AZ::Interface<FocusModeInterface>::Get();
+        EXPECT_TRUE(focusModeInterface != nullptr);
+
+        focusModeInterface->SetFocusRoot(m_entityMap["carId"]);
+        EXPECT_EQ(focusModeInterface->GetFocusRoot(), m_entityMap["carId"]);
+
+        focusModeInterface->ClearFocusRoot();
+        EXPECT_EQ(focusModeInterface->GetFocusRoot(), AZ::EntityId());
+    }
+
+    TEST_F(EditorFocusModeTests, EditorFocusModeTests_IsInFocusSubTree)
+    {
+        FocusModeInterface* focusModeInterface = AZ::Interface<FocusModeInterface>::Get();
+        EXPECT_TRUE(focusModeInterface != nullptr);
+
+        focusModeInterface->ClearFocusRoot();
+        
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["cityId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["streetId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["carId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["passengerId1"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["sportsCarId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["passengerId2"]), true);
+
+        focusModeInterface->SetFocusRoot(m_entityMap["streetId"]);
+
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["cityId"]), false);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["streetId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["carId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["passengerId1"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["sportsCarId"]), true);
+        EXPECT_EQ(focusModeInterface->IsInFocusSubTree(m_entityMap["passengerId2"]), true);
```

Comment:
suggestion: Moving the strings to constants might be better if we ever want to rename them (also this test is checking quite a few things at once, having more precise naming for the test could help clarify what behavior we're verifying)